docs/src/main/tut/deployment.md

```diff
@@ -0,0 +1,148 @@
+
+After we have built the assembled JAR containing all our JAVA dependencies, we use that JAR to build our native image. In the command below we need to replace the muslC and assembly jar paths with the appropriate locations.
+
+```
+native-image --static -H:+ReportExceptionStackTraces -H:UseMuslC="/path.to/muslC" --allow-incomplete-classpath --no-fallback --initialize-at-build-time --enable-http --enable-https --enable-all-security-services --verbose -jar "./path.to.assembly.jar" projectBinaryImage
+```
+
+A breakout for the command parameters (image generation options) :
+
+`-H:UseMuslC` to fix DNS and related segfaults and build a true static image that doesn't depend on linked libraries. 
+
+`--initialize-at-build-time` again is related to building the static image. Again we lose performance if we instead do this at runtime.
+
+`--enable-http`, `--enable-https` should correspond to related protocols
+
+`--enable-all-security-services` will likely be required to make or receive https requests.
+
+Additional image generation options are found in the [native image reference](https://www.graalvm.org/docs/reference-manual/native-image/)
+
+
+#### Execute the native image
+
+Finally we should be able to execute our output `projectBinaryImage`
+
+```
+./projectBinaryImage
+```
+
+At this point we may want to package the binary in a `docker` container, integrate the image generation to a greater build process, create `init` scripts via `systemd`, etc.
+
+#### Why static? 
+
+As an alternative to executing via the `JVM` `GraalVM native-image` allows us to execute as a native binary. For example in Linux environments this might be known as an `ELF` format. There are a number of ways to generate a native image be it `dynamic` or `static`. 
+
+A `dynamic` image is less portable because it depends on shared libary files on each linux host. Then similar to creating an assembly jar containing all our java dependencies, when we build a `static` native image, we build
+a more portable assembly including all the dependencies to run our binary across multiple platforms. For example we could expect a `static` ELF-64 binary to work across multiple linux distros of different versions for the same architecture. 
+
+
+```tut:invisible
+blockingPool.shutdown()
+```
+
+[service]: ../service
+[entity]: ../entity
+[json]: ../json
+[`ContextShift`]: https://typelevel.org/cats-effect/datatypes/contextshift.html
+[`ConcurrentEffect`]: https://typelevel.org/cats-effect/typeclasses/concurrent-effect.html
+[`IOApp`]: https://typelevel.org/cats-effect/datatypes/ioapp.html
+[middleware]: ../middleware
+[Follow Redirect]: ../api/org/http4s/client/middleware/FollowRedirect$
+[Retry]: ../api/org/http4s/client/middleware/Retry$
+[Metrics]: ../api/org/http4s/client/middleware/Metrics$
+[Request Logger]: ../api/org/http4s/client/middleware/RequestLogger$
+[Response Logger]: ../api/org/http4s/client/middleware/ResponseLogger$
+[Logger]: ../api/org/http4s/client/middleware/Logger$
```

Comment:
Most (all?) of these are unreferenced above.  Like the blockingPool, this might be copy-paste from another tut.

Reply:
Yes, I wasn't sure if that was an index for the other tutorial sections or what.

Reply:
Let's remove unnecessary.

Reply:
Yep, this is another thing that's file local.

Reply:
removed